cirq-google/cirq_google/api/v2/program.proto

```diff
@@ -431,11 +431,31 @@ message ArgMapping {
   repeated ArgEntry entries = 1;
 }
 
+message FunctionInterpolationData {
+  // If the function encoded has dimension `d`, then |x| = d |y|. Where |x| the length of `x`.
+  repeated float x = 1 [packed = true];  // The independent variable.
+  repeated float y = 2 [packed = true];  // The dependent variable.
+
+  // The interpolation method.
+  // Currently only "interp" (i.e numpy.interp) is supported.
+  string method = 3;
+}
+
+message CustomArg {
+  oneof custom_arg {
+    FunctionInterpolationData function_interpolation_data = 1;
+  }
+}
+
 message InternalGate{
   string name = 1;  // Gate name.
   string module = 2;  // Gate module. 
   int32 num_qubits = 3;  // Number of qubits. Required during deserialization.
   map<string, Arg> gate_args = 4;  // Gate args.
+
+  // Custom args are arguments that need require special processing during
+  // deserialization.
+  map<string, CustomArg> custom_args = 5;
```

Comment:
Please document what the `string` keys in this map represent.

Reply:
done